synapse/tests/test_lib_net.py

```diff
@@ -2,8 +2,198 @@
+        with self.raises(AttributeError) as e:
+            [logger.error(item) for item in chan.iter()]
+
+        chan.setq()
+        msgs = (
+            ('woo1', {}),
+            ('woo2', {}),
+            ('woo3', {}),
+            ('woo4', {}),
+            ('woo5', {}),
+        )
+
+        [chan.rx(None, msg) for msg in msgs]
+        self.eq(chan.next(timeout=1), msgs[0])
+        self.eq(chan.next(timeout=1), msgs[1])
+        self.eq(chan.next(timeout=1), msgs[2])
+        self.eq(chan.next(timeout=1), msgs[3])
+        self.eq(chan.next(timeout=1), msgs[4])
+        self.none(chan.next(timeout=1))
+
+        [chan.rx(None, msg) for msg in msgs]
+        self.eq(chan.slice(4), [msgs[0], msgs[1], msgs[2], msgs[3]])
+        self.eq(chan.next(timeout=1), msgs[4])
+        self.none(chan.next(timeout=1))
+        self.none(chan.slice(100, timeout=1))
+
+        results = []
+        [chan.rx(None, msg) for msg in msgs]
+        [results.append(item) for item in chan.iter(timeout=1)]
+        self.eq(results, list(msgs))
+        [results.append(item) for item in chan.iter(timeout=1)]
+        self.eq(results, list(msgs))
+
+        self.false(chan._chan_rxq._que_done)
+        chan.rxfini()
+        self.true(chan._chan_rxq._que_done)
+        [chan.rx(None, msg) for msg in msgs]
+        self.none(chan.next(timeout=1))
+
+    def test_lib_net_link_tx(self):
+        class DstLink(s_net.Link):
+            def __init__(self):
+                s_net.Link.__init__(self)
+                self.callcount = 0
+            def tx(self, mesg):
+                self.callcount += 1
+
+        dstlink = DstLink()
+        link = s_net.Link(link=dstlink)
+        self.none(link.txtime)
+        self.eq(dstlink.callcount, 0)
+
+        link.tx(('cool', {}))
+        first_txtime = link.txtime
+        self.ge(first_txtime, 0)
+        self.eq(dstlink.callcount, 1)
+
+        link.rxfini()
+        link.txfini(data=('cool', {}))
+        second_txtime = link.txtime
+        self.ge(second_txtime, first_txtime)
+        self.eq(dstlink.callcount, 2)
+
+        link.tx(('cool', {}))
+        self.eq(link.txtime, second_txtime)
+        self.eq(dstlink.callcount, 2)
+
+    def test_lib_net_link_repr(self):
+        link = s_net.Link()
+        rep = link.__repr__()
+        self.len(28, rep)
+        self.true(rep.startswith('Link: None at 0x'))
+
+    def test_lib_net_link_props(self):
+        link = s_net.Link()
+        self.none(link.getLinkProp('nope'))
+        self.eq(link.getLinkProp('nope', defval=1337), 1337)
+        self.none(link.setLinkProp('nope', 1337))
+        self.eq(link.getLinkProp('nope'), 1337)
+
+    def test_lib_net_link_onrx(self):
+        msg = 'woohoo'
+        def fn(self, *args, **kwargs):
+            logger.error('woohoo')
+
+        link = s_net.Link()
+        link.onrx(fn)
+        self.eq(link.rxfunc, fn)
+
+        with self.getLoggerStream('synapse.tests.test_lib_net', msg) as stream:
```

Comment:
in alot of cases with ``getLoggerStream`` you're providing the second parameter which will cause the streamevent to be set when the message occurs; but not always waiting for it. is there a reason for that? do most of the tests just run in mainthread?

Reply:
also i would operate on the stream to read the messages outside of the ``with`` block to ensure that the stream has been removed from the logger and nothing more is written to it

Reply:
No. I should fix this so that all the tests wait

Reply:
fixed